compiler/optimizer/SwitchAnalyzer.cpp

```diff
@@ -1343,7 +1343,7 @@ int32_t *TR::SwitchAnalyzer::setupFrequencies(TR::Node *node)
       TR::Node *caseNode = node->getChild(i);
       TR::Block *targetBlock = caseNode->getBranchDestination()->getNode()->getBlock();
       int32_t targetCount = targetCounts[targetBlock->getNumber()];
-      TR_ASSERT(targetCount != 0, "unreachle successor of switch statement");
+      TR_ASSERT_FATAL(targetCount > 0, "Successor block_%d of switch statement has non-sense successsor count", targetBlock->getNumber());
```

Comment:
Would it be useful to also print `targetCount` as part of the assert message?

Reply:
I don't think it is materially interesting... knowing which block should be sufficient to rerun the algorithm by hand on a jitdump to see the problem - it was the one piece of information I needed to debug the problem OpenJ9 hit because of this.